### PR TITLE
[PM-1095][PM-1104] Update email template

### PR DESCRIPTION
**Pull request by andrebispo5**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Update styling of email templates for **User Organization Invite** and **User Organization Confirmed**.

## Code changes

* **src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs:** 
Added open sans font. Added css styling. Body block shows has having changes but nothing was changed there, code formatter made the changes.

* **src/Core/MailTemplates/Handlebars/Layouts/TitleContactUs.html.hbs:** 
New base layout for title and contact us sections. The title is break in three parts for reusability purposes. Middle is Bold. Html injection is not possible because it would appear in the simple text format.

* **src/Core/MailTemplates/Handlebars/OrganizationUserConfirmed.html.hbs:**
* **src/Core/MailTemplates/Handlebars/OrganizationUserInvited.html.hbs:**
Updated UI and text. 

* **src/Core/Services/IMailService.cs:**
Changed Method signature to have `isFreeOrg` argument. We need this to make the email title different for free organizations.
 
## Screenshots
image
image
image

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/MailTemplates/Handlebars/Layouts/TitleContactUs.html.hbs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿{{#>FullHtmlLayout}}
+<div width="auto" cellpadding="0" cellspacing="0" style="padding:0; margin:-20px">
+    <div style="display: block; min-height: 204px; background-color: #175DDC; margin: 0;">
+        <div class="title-header-text" >
+            <div style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; box-sizing: border-box; font-size: 24px; color: #ffffff; line-height: 32px; font-weight: 400; -webkit-font-smoothing: antialiased; -webkit-text-size-adjust: none;">
+                {{TitleFirst}}<b class="white-title">{{TitleSecondBold}}</b>{{TitleThird}}
+            </div>
+        </div>
+        <div class="title-header-image" style="margin-left: auto; margin-right: 0px; vertical-align: bottom;">
+            <img style="margin-left: auto; margin-right: 0px; display: block;" alt='' src='https://assets.bitwarden.com/email/v1/business.png' />
```
Comment (jlf0dev) [inline src/Core/MailTemplates/Handlebars/Layouts/TitleContactUs.html.hbs:10]: Issue: We likely need to add alternate descriptions to the image so they can at least display text when they can't load

File: src/Core/MailTemplates/Handlebars/Layouts/TitleContactUs.html.hbs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <div class="footer-text" style="display: inline-block; font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; font-style: normal; font-weight: 400; font-size: 16px; line-height: 24px; padding-left: 35px; padding-right: 35px; margin-top: 15px; margin-bottom: 20px; margin-left: auto; margin-right: auto; vertical-align: middle; ">
+            <p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; font-style: normal; font-weight: 600; font-size: 20px; line-height: 28px;">We’re here for you!</p>
+            If you have any questions, search the Bitwarden <a style="text-decoration: none; color: #175DDC; font-weight: 600;" href="https://bitwarden.com/help/">Help</a> site or <a style="text-decoration: none; color: #175DDC; font-weight: 600;" href="https://bitwarden.com/contact/">contact us</a>.
+        </div>
+        <div style="display: inline-block; width:33%; margin-left: auto; margin-right: auto; vertical-align: middle; ">
+            <img class="footer-image" src="https://assets.bitwarden.com/email/v1/chat.png" style="width: 94.73px; height: 77.25px; margin-left: auto; margin-right: 30px;" alt="" />
```
Comment (jlf0dev) [inline src/Core/MailTemplates/Handlebars/Layouts/TitleContactUs.html.hbs:22]: Issue: Another alt description

File: src/Core/Services/Implementations/HandlebarsMailService.cs [hunk 3]
```diff
@@ -200,21 +203,24 @@ public async Task SendOrganizationConfirmedEmailAsync(string organizationName, s
...
-    public async Task BulkSendOrganizationInviteEmailAsync(string organizationName, IEnumerable<(OrganizationUser orgUser, ExpiringToken token)> invites)
+    public async Task BulkSendOrganizationInviteEmailAsync(string organizationName, IEnumerable<(OrganizationUser orgUser, ExpiringToken token)> invites, bool isFreeOrg)
     {
         MailQueueMessage CreateMessage(string email, object model)
         {
             var message = CreateDefaultMessage($"Join {organizationName}", email);
             return new MailQueueMessage(message, "OrganizationUserInvited", model);
         }
-
+        var freeOrgTitle = CoreHelpers.SanitizeForEmail("A Bitwarden member invited you to an organization. Join now to start securing your passwords!", false);
```
Comment (jlf0dev) [inline src/Core/Services/Implementations/HandlebarsMailService.cs:216]: Question: Does this string need to be sanitized?

**Review by jlf0dev — CHANGES_REQUESTED** [review]
These templates are looking great! I initially missed your comment about the formatter on `Full.html.hbs` so I added a bunch of comments. You can ignore those if you'd like, or address if them if you have the time, it's up to you.

**Comment by andrebispo5** [comment]
@jlf0dev We will need to create a tech dept item to refactor the `Full.html.hbs`to improve the nested tables and other parts. It is being used in all emails and i didn't want to break anything at this time, the only thing i changed there or added were styles. Also i didn't add the alt text because in some cases it will hide the images completely, and it was the initial desired behaviour but since it won't work the same in all clients i'll add the alts.

File: src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs [hunk 4]
```diff
@@ -45,9 +44,20 @@
 
         body {
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs:45]: To be reviewed in tech dept.

File: src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs [hunk 5]
```diff
@@ -119,54 +172,58 @@
...
-    </td></tr></table>
+    <table width="100%" cellpadding="0" cellspacing="0" bgcolor="#f6f6f6">
+        <tr>
+            <td class="container" width="100%" align="center">
+                {{! 600px container }}
+                <table cellpadding="0" cellspacing="0" width="100%" class="content-table">
+                    <tr>
+                        <td></td> {{! Left column (center fix) }}
+                        <td class="content" align="center" valign="top" width="600" style="padding-bottom: 20px;">
+                            <table class="header" cellpadding="0" cellspacing="0" width="100%">
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs:183]: To be reviewed in tech dept.

File: src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs [hunk 6]
```diff
@@ -119,54 +172,58 @@
...
+                                <tr>
+                                    <td valign="middle" class="aligncenter middle logo" style="padding: 20px 0 10px;" align="center">
+                                        <img src="https://bitwarden.com/images/logo-horizontal-blue.png" alt="" width="250" height="39" />
+                                    </td>
+                                </tr>
+                            </table>
+                            <table class="main" cellpadding="0" cellspacing="0" style="border: 1px solid #e9e9e9; border-radius: 3px;" bgcolor="white">
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs:190]: To be reviewed in tech dept.

File: src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs [hunk 7]
```diff
@@ -119,54 +172,58 @@
...
+                                    <td class="content-wrap" style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; box-sizing: border-box; font-size: 16px; color: #333; line-height: 25px; margin: 0; -webkit-font-smoothing: antialiased; padding: 20px; -webkit-text-size-adjust: none;" valign="top">
+
+                                        {{>@partial-block}}
+
+                                    </td>
+                                </tr>
+                            </table>
+                            <table class="footer" cellpadding="0" cellspacing="0" width="100%" style="margin: 0; width: 100%;">
+                                <tr>
+                                    <td class="aligncenter social-icons" align="center" style="margin: 0; padding: 15px 0 0 0;" valign="top">
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs:201]: To be reviewed in tech dept.

File: src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs [hunk 8]
```diff
@@ -119,54 +172,58 @@
...
+                                        <table cellpadding="0" cellspacing="0" style="margin: 0 auto;">
+                                            <tr>
+                                                <td style="margin: 0; padding: 0 10px;" valign="top"><a href="https://twitter.com/bitwarden" target="_blank"><img src="https://bitwarden.com/images/mail-twitter.png" alt="Twitter" width="30" height="30" /></a></td>
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs:204]: To be reviewed in tech dept.

File: src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs [hunk 9]
```diff
@@ -119,54 +172,58 @@
...
+                                                <td style="margin: 0; padding: 0 10px;" valign="top"><a href="https://www.linkedin.com/company/bitwarden1/" target="_blank"><img src="https://bitwarden.com/images/mail-linkedin.png" alt="LinkedIn" width="30" height="30" /></a></td>
+                                                <td style="margin: 0; padding: 0 10px;" valign="top"><a href="https://www.facebook.com/bitwarden/" target="_blank"><img src="https://bitwarden.com/images/mail-facebook.png" alt="Facebook" width="30" height="30" /></a></td>
+                                            </tr>
+                                        </table>
+                                    </td>
+                                </tr>
+                                <tr>
+                                    <td class="content-block" style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; box-sizing: border-box; font-size: 14px; color: #666666; line-height: 25px; margin: 0; -webkit-font-smoothing: antialiased; padding: 15px 0 0 0; -webkit-text-size-adjust: none; text-align: center;" valign="top">
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply src/Core/MailTemplates/Handlebars/Layouts/Full.html.hbs:216]: To be reviewed in tech dept.

File: src/Core/Services/Implementations/HandlebarsMailService.cs (hunk 3 above)
Comment (andrebispo5) [reply src/Core/Services/Implementations/HandlebarsMailService.cs:216]: No it does not, i added it when i was testing another thing and forgot to remove it, thanks!

File: src/Core/MailTemplates/Handlebars/Layouts/TitleContactUs.html.hbs (hunk 1 above)
Comment (andrebispo5) [reply src/Core/MailTemplates/Handlebars/Layouts/TitleContactUs.html.hbs:10]: I added a comment to the PR saying that i would add the alt text but after reading a little about it, the best practice for decorative images it to leave them as empty. 
Ref: 
https://accessibility.huit.harvard.edu/describe-content-images 
https://help.siteimprove.com/support/solutions/articles/80000863904-accessibility-image-alt-text-best-practices

File: src/Core/MailTemplates/Handlebars/Layouts/TitleContactUs.html.hbs (hunk 2 above)
Comment (andrebispo5) [reply src/Core/MailTemplates/Handlebars/Layouts/TitleContactUs.html.hbs:22]: I added a comment to the PR saying that i would add the alt text but after reading a little about it, the best practice for decorative images it to leave them as empty.
Ref:
https://accessibility.huit.harvard.edu/describe-content-images
https://help.siteimprove.com/support/solutions/articles/80000863904-accessibility-image-alt-text-best-practices

**Review by jlf0dev — APPROVED** [review]
Everything looks good! I went ahead and made a [tech debt task](https://bitwarden.atlassian.net/browse/PM-1413) for the `Full.html.hbs` refactor 👍

**Pull request merged**